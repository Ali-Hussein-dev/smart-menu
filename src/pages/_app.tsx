import { AppProps } from 'next/app'
import { ChakraProvider } from '@chakra-ui/react'
import '../css/global.css'
import Head from 'next/head'

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export default function MyApp({ Component, pageProps }: AppProps) {
  return (
    <ChakraProvider>
      <Head>
        <title>Smart Gastkarte</title>
        {/* <link rel="icon" href="/favicon.ico" /> */}
      </Head>
      <Component {...pageProps} />
    </ChakraProvider>
  )
}
