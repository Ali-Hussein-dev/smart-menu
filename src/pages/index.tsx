import Head from 'next/head'

export const Home = (): JSX.Element => (
  <div className="grid h-screen place-items-center bg-blueGray-100 text-blueGray-600">
    <Head>
      <title>Smart Gastkarte | HomePage</title>
      {/* <link rel="icon" href="/favicon.ico" /> */}
    </Head>
    <main>
      HomePage
    </main>
  </div>
)

export default Home
