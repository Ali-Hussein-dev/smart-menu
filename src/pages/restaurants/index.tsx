import MegaList from '@/database/index'
import { Avatar } from '@chakra-ui/react'
import Link from 'next/link'

//=======================
const RestaurantsIndex: React.FC = () => {
  // hooks
  //--------------------------------------
  // functions
  //--------------------------------------
  return (
    <div className="flex flex-col items-center min-h-screen pt-4 text-3xl bg-blueGray-200 text-blueGray-700">
      <h1 className="mb-5">Restaurants Liste</h1>
      {MegaList.map((o) => (
        <div
          className="flex items-center w-10/12 px-5 py-3 rounded-lg shadow-lg md:w-8/12 gap-x-4 bg-blueGray-50"
          key={o.name}
        >
          <Avatar src={o.meta.imgSrc} name={o.name} size="xl" />
          <Link href={`/restaurants/${o.name}`}>
            <a>{o.name}</a>
          </Link>
        </div>
      ))}
    </div>
  )
}
export default RestaurantsIndex
