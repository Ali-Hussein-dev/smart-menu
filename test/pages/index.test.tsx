import * as React from 'react'
import { render } from '../testUtils'
import { Home } from '../../src/pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { container } = render(<Home />, {})
    expect(container).toMatchSnapshot()
  })
})
export {}
