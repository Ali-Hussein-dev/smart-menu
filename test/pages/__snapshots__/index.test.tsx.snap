// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<div>
  <div
    class="grid h-screen place-items-center bg-blueGray-100 text-blueGray-600"
  >
    <main>
      HomePage
    </main>
  </div>
</div>
`;
